README.md

```diff
@@ -20,7 +20,58 @@ spago install machines
 
 ## Quick start
 
-The quick start hasn't been written yet. Contributions are welcome!
+The examples here use `Identity` for simplicity. Usually, you would use
+a different `Monad`, such as `Effect`, `Aff`, `State`, etc.
+
+There are several ways to build machines. One way is to use `do` syntax,
+for example:
+
+```purescript
+machine1 :: MealyT Identity Unit String
+machine1 = do
+  number <- fromArray [10, 20, 30, 40, 50, 0, 60, 70]
+  scaled <-
+      if number == 0
+          then halt
+          else pure $ div number 2
+  pure $ show scaled
+```
+
+This will create a machine `machine1` which goes through the "inputs"
+from the array. It then checks and halts on any zero input, and otherwise
+scales the inputs (by dividing by 2). The result is then transformed intoa string.
+
+The resulting machine can be materialized via `toUnfoldable unit machine1 :: Array String`. The result is `["5","10","15","20","25"]`.
+
+Another way to write the same machine is using machine composition:
+
+```purescript
+machine2 :: MealyT Identity Unit String
+machine2 =
+    fromArray [10, 20, 30, 40, 50, 0, 60, 70]
+        >>> pureMealy haltOn0
+        >>> pureMealy scale
+        >>> pureMealy pretty
+  where
+    haltOn0 :: Int -> Step Identity Int Int
+    haltOn0 0 = Halt
+    haltOn0 n = Emit n $ pureMealy haltOn0
+
+    scale :: Int -> Step Identity Int Int
+    scale n = Emit (n `div` 2) $ pureMealy scale
```

Comment:
For parity with the earlier example, should these two be merged?

```purs
    scale :: Int -> Step Identity Int Int
    scale n
      | n == 0 = Halt
      | otherwise = Emit (n `div` 2) $ pureMealy scale
```

Reply:
You are right! I changed the examples a bit, using `guard` in the monadic version. Does this look better?

Reply:
If you are comfortable switching to two spaces instead of four in the example (as is done in the `do` example above) I'd appreciate that, but otherwise 👍!

Reply:
I'm going to leave this unmerged to give you a moment to respond to this, but at that point I will go ahead and merge.